### DISCUSSION TITLE
Sad Synology User - Any Assistance much appreciated :)

### DISCUSSION THREAD

**Discussion by boba51** [teslamate-org/teslamate#4188] [Q&A] [answered]
After struggling to get TM up and running on my Synology NAS many years ago, I needed to wipe my NAS to convert it's file format, I took a backup of my "docker" folder naively thinking that would cover it, I now know better and have resigned myself that that data is now gone and is not the focus of this request, that is below.

Since the rebuild a few weeks now I've been trying to set up a fresh instance but whatever I try it ends up with Teslamate not being able to access the database, I've tried various installs, wiping the volumes between each attempt, I've even followed a guide to build within Container Manager as a Project as the creator of the guide has it running with no issue, I can only imagine there is something on my NAS interfering with things.

I had to change the Db port to 6432 as 5432 in use by the Synology itself, other than that everything should be standard, as it was running nicely before, but I do updated the NAS and it's Apps regularly so may be related to that and probably something really basic I've overlooked!

Status:
Last build using Synology Container Manager
Can access Grafana on Host IP:3001 - No Data
TeslaMate - No access to webpage - 40000 and Log displays cannot connect to 6432 (revised port) 

Tried:
Rebuilding from SSH numerous times - Changing "docker-compose.yml" options
Synology FW - Different entries and totally off
Any any other suggestions I've found online :(

DSM - Version 7.2-64561
Model name  DS918+
CPU  INTEL Celeron J3455
CPU clock rate 1.5 GHZ
CPU cores  4
Total physical memory  4096 MB

Any help would be greatly appreciated

Thanks
Bob

**Comment by dyxyl** [comment]
I suggest you post your complete docker-compose.yml and the corresponding docker logs for the database and teslamate containers.

**Reply by boba51** [reply]
Thanks, will do tomorrow, just off to bed and then work, I just hoped if anyone had experienced the same issue they might suggest something.
Is there a quick way to sanitise the files for passwords etc? 
Bob

**Reply by boba51** [reply]
So after another night of trying various things I'm no further here are a few file/logs for consideration:

Nothing seems to error, just TeslaMate cannot connect to the db :(

**docker-compose ps:**
NAME                 IMAGE                        COMMAND                  SERVICE             CREATED             STATUS              PORTS
docker-database-1    postgres:16                  "docker-entrypoint.s…"   database            14 minutes ago      Up 9 minutes        0.0.0.0:6432->5432/tcp, :::6432->5432/tcp
docker-grafana-1     teslamate/grafana:latest     "/run.sh"                grafana             14 minutes ago      Up 13 minutes       0.0.0.0:3001->3000/tcp, :::3001->3000/tcp
docker-mosquitto-1   eclipse-mosquitto:2          "/docker-entrypoint.…"   mosquitto           14 minutes ago      Up 13 minutes       1883/tcp
docker-teslamate-1   teslamate/teslamate:latest   "tini -- /bin/sh /en…"   teslamate           14 minutes ago      Up 13 minutes       0.0.0.0:4000->4000/tcp, :::4000->4000/tcp

**docker-compose.yml:**
services:
  teslamate:
    image: teslamate/teslamate:latest
    restart: always
    environment:
      - ENCRYPTION_KEY=redacted #replace with a secure key to encrypt your Tesla API tokens
      - DATABASE_USER=teslamate
      - DATABASE_PASS=redacted #insert your secure database password!
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
      - MQTT_HOST=mosquitto
      - DATABASE_PORT=6432
    ports:
      - 4000:4000
    volumes:
      - /../volume1/docker/teslamate/import:/opt/app/import
    cap_drop:
      - all

  database:
    image: postgres:16
    restart: always
    environment:
      - POSTGRES_USER=teslamate
      - POSTGRES_PASSWORD=redacted #insert your secure database password!
      - POSTGRES_DB=teslamate
    ports:
      - 6432:5432
    volumes:
      - teslamate-db:/var/lib/postgresql/data

  grafana:
    image: teslamate/grafana:latest
    restart: always
    environment:
      - DATABASE_USER=teslamate
      - DATABASE_PASS=redacted #insert your secure database password!
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
    ports:
      - 3001:3000
    volumes:
      - teslamate-grafana-data:/var/lib/grafana

  mosquitto:
    image: eclipse-mosquitto:2
    restart: always
    command: mosquitto -c /mosquitto-no-auth.conf
    # ports:
    #   - 1883:1883
    volumes:
      - mosquitto-conf:/mosquitto/config
      - mosquitto-data:/mosquitto/data

volumes:
  teslamate-db:
  teslamate-grafana-data:
  mosquitto-conf:
  mosquitto-data:

**docker-database-1**
date	stream	content
2024/09/09 23:32:09	stderr	2024-09-09 22:32:09.514 UTC [27] LOG:  checkpoint complete: wrote 3 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.001 s, sync=0.145 s, total=0.969 s; sync files=2, longest=0.073 s, average=0.073 s; distance=0 kB, estimate=0 kB; lsn=0/1933810, redo lsn=0/19337D8
2024/09/09 23:32:08	stderr	2024-09-09 22:32:08.545 UTC [27] LOG:  checkpoint starting: time
2024/09/09 23:27:09	stderr	2024-09-09 22:27:08.719 UTC [1] LOG:  database system is ready to accept connections
2024/09/09 23:27:08	stderr	2024-09-09 22:27:08.446 UTC [29] LOG:  database system was shut down at 2024-09-09 22:26:10 UTC
2024/09/09 23:27:08	stderr	2024-09-09 22:27:08.223 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2024/09/09 23:27:08	stderr	2024-09-09 22:27:08.154 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2024/09/09 23:27:08	stderr	2024-09-09 22:27:08.154 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2024/09/09 23:27:08	stderr	2024-09-09 22:27:08.154 UTC [1] LOG:  starting PostgreSQL 16.4 (Debian 16.4-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
2024/09/09 23:27:07	stdout	
2024/09/09 23:27:07	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/09 23:27:07	stdout	
2024/09/09 23:26:10	stderr	2024-09-09 22:26:10.608 UTC [1] LOG:  database system is shut down
2024/09/09 23:26:10	stderr	2024-09-09 22:26:10.601 UTC [27] LOG:  checkpoint complete: wrote 3 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.001 s, sync=0.266 s, total=0.940 s; sync files=2, longest=0.134 s, average=0.133 s; distance=0 kB, estimate=0 kB; lsn=0/1933728, redo lsn=0/1933728
2024/09/09 23:26:09	stderr	2024-09-09 22:26:09.757 UTC [27] LOG:  checkpoint starting: shutdown immediate
2024/09/09 23:26:09	stderr	2024-09-09 22:26:09.662 UTC [27] LOG:  shutting down
2024/09/09 23:26:09	stderr	2024-09-09 22:26:09.660 UTC [1] LOG:  background worker "logical replication launcher" (PID 32) exited with exit code 1
2024/09/09 23:26:09	stderr	2024-09-09 22:26:09.657 UTC [1] LOG:  aborting any active transactions
2024/09/09 23:26:09	stderr	2024-09-09 22:26:09.617 UTC [1] LOG:  received fast shutdown request
2024/09/09 23:22:50	stderr	2024-09-09 22:22:50.998 UTC [1] LOG:  database system is ready to accept connections
2024/09/09 23:22:50	stderr	2024-09-09 22:22:50.204 UTC [29] LOG:  database system was shut down at 2024-09-09 22:18:10 UTC
2024/09/09 23:22:49	stderr	2024-09-09 22:22:49.233 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2024/09/09 23:22:48	stderr	2024-09-09 22:22:48.209 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2024/09/09 23:22:48	stderr	2024-09-09 22:22:48.209 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2024/09/09 23:22:48	stderr	2024-09-09 22:22:48.191 UTC [1] LOG:  starting PostgreSQL 16.4 (Debian 16.4-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
2024/09/09 23:22:45	stdout	
2024/09/09 23:22:45	stdout	PostgreSQL Database directory appears to contain a database; Skipping initialization
2024/09/09 23:22:45	stdout	


**docker-teslamate-1**
date	stream	content
2024/09/09 23:49:38	stdout	waiting for postgres at database:6432
2024/09/09 23:49:37	stdout	waiting for postgres at database:6432
2024/09/09 23:49:36	stdout	waiting for postgres at database:6432
2024/09/09 23:49:35	stdout	waiting for postgres at database:6432
2024/09/09 23:49:34	stdout	waiting for postgres at database:6432
2024/09/09 23:49:33	stdout	waiting for postgres at database:6432
2024/09/09 23:49:32	stdout	waiting for postgres at database:6432
2024/09/09 23:49:31	stdout	waiting for postgres at database:6432
2024/09/09 23:49:30	stdout	waiting for postgres at database:6432
2024/09/09 23:49:29	stdout	waiting for postgres at database:6432
2024/09/09 23:49:28	stdout	waiting for postgres at database:6432
2024/09/09 23:49:27	stdout	waiting for postgres at database:6432
2024/09/09 23:49:26	stdout	waiting for postgres at database:6432
2024/09/09 23:49:25	stdout	waiting for postgres at database:6432
2024/09/09 23:49:24	stdout	waiting for postgres at database:6432
2024/09/09 23:49:23	stdout	waiting for postgres at database:6432
repeated many times

**Comment by andrewjw** [comment, marked as answer]
Try removing the `DATABASE_PORT=6432` from the Teslamate section. You've redirected the port for outside the docker environment, but internally (which is what Teslamate will be using) it's still the default.

Hope that helps,
Andrew

**Reply by boba51** [reply]
Hi Andrew,

To be honest not sure what's changed, but when I originally tried to run TM the database came up and said it couldn't use 5432 as already in use, hence me going down the defining alternative ports, but tonight I stripped out the Port commands and TM has connected again, now I just need to do the Token stuff.

Thanks for your help, not sure how I got to where I was but learnt a lot and should be good to go again, many thanks!
Bob

**Comment by boba51** [comment]
Hi Andrew,
Thanks for your suggestion :)

I can't use port 5432 as already in use on the NAS by something else, my understanding (mainly limited to trying to resolve this issue, so not vast) is DATABASE_PORT=6432 instructs TM to use this port, confirmed by log entry - 2024/09/09 23:49:38 stdout waiting for postgres at database:6432

And the corresponding ports: - 6432:5432 under the db section should listen on 6432 and redirect to 5432 (confirmed by the PS output - docker-database-1 postgres:16 "docker-entrypoint.s…" database 14 minutes ago Up 9 minutes 0.0.0.0:6432->5432/tcp, :::6432->5432/tcp

But I have also tried various combinations of these ports and any changes from the default yaml file are attempts at fixing the issue unfortunately all have met with the same result, but happy to try something if someone has a working config on Synology DSM 7.2 (I'm wondering if the newish Container Manager App changed/broke anything?)

Thanks again,
Bob

**Reply by andrewjw** [reply]
The port redirection is for outside the Docker network. As Teslamate is running inside the Docker network it doesn't apply, and from its perspective, Postgres is listening on 5432.

My Teslamate is running on DSM 7.2 with no problems. You can see my docker-compose file here - https://github.com/andrewjw/docker/blob/master/docker-compose.yaml. I have a lot of containers running, so there is a lot of extra noise in the file I'm afraid.

If it still doesn't work after telling Teslamate to connect to the database on 5432, it might be worth posting the output from the standard docker-compose file.

Andrew

**Reply by boba51** [reply]
Thanks @andrewjw,

I'll take another look tonight, but the db won't start configured with 5432 as the port is already in use and as mentioned TM failed originally with TM looking for the db at 5432, that's when I added the DATABASE_PORT=6432.

But thinking if ports only for external access in why is the TM "DATABASE_PORT" environment variable available?

Thanks
Bob
